apps/sms/services/js/threads.js

```diff
@@ -6,16 +6,24 @@
   var threads = new Map();
   var messageMap = new Map();
 
-  function Thread(thread) {
-    var length = Thread.FIELDS.length;
-    var key;
-
-    for (var i = 0; i < length; i++) {
-      key = Thread.FIELDS[i];
-      this[key] = thread[key];
-    }
+  function extractSupportedFields(
+    { id, body, timestamp, participants, lastMessageType, status, isDraft }
+  ) {
+    return {
```

Comment:
I remember @julienw was against "lastMessageSubject" since we don't use it, so I removed it in this patch :) I can return it back if you guys think it's necessary.